pkg/provider/azure_vmss_cache.go

```diff
@@ -142,25 +136,25 @@ func extractVmssVMName(name string) (string, string, error) {
 	return ssName, instanceID, nil
 }
 
-// getVMSSVMCache returns an *azcache.TimedCache and cache key for a VMSS (creating that cache if new).
-func (ss *ScaleSet) getVMSSVMCache(resourceGroup, vmssName string) (string, *azcache.TimedCache, error) {
-	cacheKey := strings.ToLower(fmt.Sprintf("%s/%s", resourceGroup, vmssName))
-	if entry, ok := ss.vmssVMCache.Load(cacheKey); ok {
-		return cacheKey, entry.(*azcache.TimedCache), nil
-	}
+func (ss *ScaleSet) getVMSSVMCacheKey(resourceGroup, vmssName string) string {
```

Comment:
move this `getVMSSVMCacheKey` to azure_utils.go is better since this func does not need to a func of `(ss *ScaleSet)`

Reply:
and also add a ut

Reply:
make sure it's `cacheKey := strings.ToLower(fmt.Sprintf("%s/%s", resourceGroup, vmssName))`  case insensitive, and lower and upper case in the ut.